# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../openfst/gcc43.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- fst/bin/main.cc.vanilla	2009-02-12 10:10:41.000000000 +0100
+++ fst/bin/main.cc	2009-02-12 10:11:12.000000000 +0100
@@ -19,6 +19,8 @@
 // functions that support multiple and extensible arc types.
 
 #include <iostream>
+#include <climits>
+#include <string.h>
 
 #include "fst/bin/main.h"
 
--- fst/lib/randgen.h.vanilla	2009-02-12 10:16:39.000000000 +0100
+++ fst/lib/randgen.h	2009-02-12 10:16:46.000000000 +0100
@@ -23,6 +23,7 @@
 #include <cmath>
 #include <cstdlib>
 #include <ctime>
+#include <climits>
 
 #include "fst/lib/mutable-fst.h"
 
--- fst/lib/symbol-table.cc.vanilla	2009-02-12 10:09:50.000000000 +0100
+++ fst/lib/symbol-table.cc	2009-02-12 10:09:59.000000000 +0100
@@ -20,6 +20,8 @@
 #include "fst/lib/symbol-table.h"
 #include "fst/lib/util.h"
 
+#include <string.h>
+
 DEFINE_bool(fst_compat_symbols, true,
             "Require symbol tables to match when appropriate");
 
--- fst/lib/compat.h.vanilla	2009-02-12 10:11:27.000000000 +0100
+++ fst/lib/compat.h	2009-02-12 10:11:49.000000000 +0100
@@ -34,6 +34,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <string.h>
 
 // exact size types
 typedef short int16;
--- fst/lib/vector-fst.h.vanilla	2009-02-12 10:10:14.000000000 +0100
+++ fst/lib/vector-fst.h	2009-02-12 10:10:21.000000000 +0100
@@ -24,6 +24,8 @@
 #include "fst/lib/mutable-fst.h"
 #include "fst/lib/test-properties.h"
 
+#include <string.h>
+
 namespace fst {
 
 template <class A> class VectorFst;
